rfcs/2021-11-03-9862-ingest-apm-stats-along-traces-in-dd-agent-source.md

```diff
@@ -0,0 +1,309 @@
+
+This also raises the question of having a second (assuming that the `datadog_agent` sources accepts Datadog Agent
+metrics - [RFC][dd-agent-metric-rfc]), unrelated, metric stream coming out of the `datadog_agent` source. The
+Vector event ingested representing APM stats will have to be routed along with traces, and most often they will follow a
+different path that other plain metrics/logs received from a `core-agent`. Thus it is suggested to re-arrange the
+`datadog_agent` source, many option are available (Additional details on what exactly are "Datadog Agents" can be
+found in the trace support [RFC][trace-support-pr] and may provide relevant context for undermentionned points):
+
+* Keeping a single `datadog_agent` source:
+  * and add a settings to switch between agent kind: `agent: <TYPE>` where `<TYPE>`
+  could be  `core` (would support metrics & logs - we could optionally add `logs` & `metrics` to only allow logs or
+  metrics, along with `core` that would allow both), `trace` and could be extended to support `process`, `security` and
+  so on
+  * or introduce the ability for a source to have multiple output for the `datadog_agent` source, e.g. <SRC_ID>.metrics, <SRC_ID>.logs, <SRC_ID>.traces, <SRC_ID>.apm_stats, etc.
+* Another solution would be to keep one Vector source per kind of Datadog Agent an then we would have the following
+  Vector sources:
+  * `datadog_core` or the current `datadog_agent` sources that would receive the data sent by the Datadog "core" Agent
+    (the agent that collects logs & metrics).
+  * `datadog_trace` that would support all data sent by the `trace-agent`
+  * And so on as the support list grows:
+    * `datadog_process` source for the Datadog `process-agent`
+    * `datadog_security` for the Datadog security Agent
+    * Etc.
+
+[apm-stats-proto]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/pb/stats.proto
+[client-side-marker]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/pb/stats.proto#L15
+[concentrator]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/stats/concentrator.go
+[client-stats-aggregator]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/stats/client_stats_aggregator.go#L25-L45
+[stats-writer]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/writer/stats.go
+[client-grouped-stats-proto]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/pb/stats.proto#L55-L70
+[sketch-pr]: https://github.com/vectordotdev/vector/pull/9178
+[call-to-toproto]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/stats/statsraw.go#L53-L62
+[ddsktech-proto]: https://github.com/DataDog/sketches-go/blob/0829b5a6a3c9627d1faee6c690611067ee110e38/ddsketch/pb/ddsketch.proto
+[vector-sketch]: https://github.com/vectordotdev/vector/blob/4db23b3/lib/vector-core/src/metrics/ddsketch.rs#L202-L213
+[dd-agent-metric-rfc]: https://github.com/vectordotdev/vector/pull/9057
+
+## Cross cutting concerns
```

Comment:
This is another place I would mention the upcoming schemas support for interoperability.

Reply:
Will do.